packages/material-ui-system/src/spacing.js

```diff
@@ -0,0 +1,102 @@
+import resonsivePropType from './resonsivePropType';
+import merge from './merge';
+
+const properties = {
+  m: 'margin',
+  p: 'padding',
+};
+
+const directions = {
+  t: 'Top',
+  r: 'Right',
+  b: 'Bottom',
+  l: 'Left',
+  x: ['Left', 'Right'],
+  y: ['Top', 'Bottom'],
+};
+
+function getCssProperties(prop) {
+  const [a, b] = prop.split('');
+  const property = properties[a];
+  const direction = directions[b] || '';
+  return Array.isArray(direction) ? direction.map(dir => property + dir) : [property + direction];
+}
+
+const spacingKeys = [
+  'm',
+  'mt',
+  'mr',
+  'mb',
+  'ml',
+  'mx',
+  'my',
+  'p',
+  'pt',
+  'pr',
+  'pb',
+  'pl',
+  'px',
+  'py',
+];
+
+function getValue(theme, propValue) {
+  const abs = Math.abs(propValue);
+  const themeTransformer = theme.spacing.unit ? theme.spacing.unit : theme.spacing;
```

Comment:
Is this just shorthand for nully check or do you actually want to fallback to `theme.spacing` if `theme.spacing.unit === 0`?

Reply:
The goal is to support `theme.spacing = 8`.

Reply:
I don't understand that statement at all. Should it be configurable or not? If so why disallow 0 spacing?

Reply:
So "yes". The spacing is not supposed to be zero. Because later you use `theme.spacing || 8` which will fallback to `8` if it is `0`.